File: lib/rust/tests/spec/route.rs
use zgz_transport::Route;

#[test]
fn names_routes_like_the_operator() {
	assert_eq!(
		[11, 14, 31, 111, 117, 210].map(|route| Route(route).to_string()),
		["Ci1", "Ci4", "31", "N1", "N7", "L1"]
	);
}

#[test]
fn keeps_numbers_around_the_named_ranges() {
	assert_eq!(
		[0, 10, 15, 110, 118, 255].map(|route| Route(route).to_string()),
		["0", "10", "15", "110", "118", "255"]
	);
	assert_eq!(Route::TRAM, Route(210));
	assert!(Route(31) < Route::TRAM);
}

File: lib/java/src/main/java/dev/hloth/zgztransport/Route.java
package dev.hloth.zgztransport;

/**
 * Route id of the operator's GTFS feed, byte 7 of a transaction.
 *
 * @param id
 *            the route id; {@code 0} on top ups made off board
 */
public record Route(int id) {

	/** The tram, printed as {@code L1}. */
	public static final Route TRAM = new Route(210);

	public Route {
		Bytes.checkRange("route", id, 0, 0xff);
	}

	/**
	 * The name the operator prints: buses keep their number, {@code 11} to
	 * {@code 14} are {@code Ci1} to {@code Ci4}, {@code 111} to {@code 117} are
	 * {@code N1} to {@code N7} and the tram is {@code L1}.
	 *
	 * @return the public name of this route
	 */
	@Override
	public String toString() {
		if (id == TRAM.id) {
			return "L1";
		}
		if (id >= 11 && id <= 14) {
			return "Ci" + (id - 10);
		}
		if (id >= 111 && id <= 117) {
			return "N" + (id - 110);
		}
		return Integer.toString(id);
	}
}
